node/heartbeat/heartbeatMessageInfo.go

```diff
@@ -192,6 +195,5 @@ func (hbmi *heartbeatMessageInfo) GetIsActive() bool {
 func (hbmi *heartbeatMessageInfo) GetIsValidator() bool {
 	hbmi.updateMutex.Lock()
 	defer hbmi.updateMutex.Unlock()
-	isValidator := hbmi.isValidator
-	return isValidator
+	return hbmi.peerType == string(core.EligibleList) || hbmi.peerType == string(core.WaitingList)
```

Comment:
can you please add a unit test for GetIsValidator? to see that the isValidator works with peerType eligible, waiting and another dummy list

Reply:
added 3 unit tests

Reply:
ok